data_steward/analytics/cdr_ops/controlled_tier_qc/check_controlled_tier_part2.py

```diff
@@ -1350,21 +1340,104 @@ def query_template(table_era):
   COUNT(*) AS bad_rows
 FROM
   `{{project_id}}.{{ct_dataset}}.survey_conduct` sc
-WHERE sc.survey_concept_id IN (2100000011,2100000012) 
+WHERE sc.survey_concept_id IN (2100000011,2100000012)
 GROUP BY 1
 """)
 q = query.render(project_id=project_id,
-            ct_dataset=ct_dataset,
-            wear_codes=WEAR_SURVEY_CODES)
-df1=execute(client, q) 
-if df1['bad_rows'].sum()==0:
- df = df.append({'query' : 'Query18 wear_consent records are cleaned as expected.', 'result' : 'PASS'},  
-                ignore_index = True) 
+                 ct_dataset=ct_dataset,
+                 wear_codes=WEAR_SURVEY_CODES)
+df1 = execute(client, q)
+if df1['bad_rows'].sum() == 0:
+    df = df.append(
+        {
+            'query': 'Query18 wear_consent records are cleaned as expected.',
+            'result': 'PASS'
+        },
+        ignore_index=True)
 else:
- df = df.append({'query' : 'Query18 wear_consent records have not been cleaned as expected.', 'result' : 'Failure'},
-                ignore_index = True) 
+    df = df.append(
+        {
+            'query':
+                'Query18 wear_consent records have not been cleaned as expected.',
+            'result':
+                'Failure'
+        },
+        ignore_index=True)
 df1
 
+# +
+# Query 19: verify all race/ethnicity values are top-level, not subcategories
+# -
+
+query = JINJA_ENV.from_string("""
+SELECT ethnicity_concept_id, ethnicity_source_concept_id, ethnicity_source_value, count(*)
+FROM `{{ project_id }}.{{ ct_dataset }}.person`
+WHERE ethnicity_concept_id NOT IN (38003563, 38003564, 1586148, 903079, 903096, 0)
+AND ethnicity_source_concept_id NOT IN (38003563, 38003564, 1586148, 903079, 903096, 0)
+GROUP BY 1, 2, 3
+
+UNION ALL
+
+SELECT race_concept_id, race_source_concept_id, race_source_value, count(*)
+FROM `{{ project_id }}.{{ct_dataset}}.person`
+WHERE race_concept_id NOT IN (2000000008, 1177221, 45882607, 903096, 0)
+AND race_source_concept_id NOT IN (1586141, 1586142, 1586143, 1586144, 1586145, 1586146, 903079, 0)
+GROUP BY 1, 2, 3
+""")
+q = query.render(project_id=project_id, ct_dataset=ct_dataset)
+result = execute(client, q)
+if result.empty:
+    df = df.append(
+        {
+            'query': 'Query19 All race/ethnicity categories are top-level',
+            'result': 'PASS'
+        },
+        ignore_index=True)
+else:
+    df = df.append(
+        {
+            'query': 'A subcategory exists for race/ethnicity',
+            'result': 'Failure'
+        },
+        ignore_index=True)
+result
+
+# Query 20:  Check for the existence of at least one race/ethnicity sub-categories
+# -
+
+query = JINJA_ENV.from_string("""
```

Comment:
This check is already merged in why do you have this again here, am i missing anything?

Reply:
This conflict is fixed